news-explorer/src/utils/MainApi.js

```diff
@@ -0,0 +1,77 @@
+import { MAIN_API_BASE_URL } from "./constants";
+
+  class MainApi {
+    constructor(baseUrl, headers) {
+      this._baseUrl = baseUrl;
+      this._headers = headers;
+      this.fetchCall = this.fetchCall.bind(this);
+    }
+
+    fetchCall(url, headers) {
```

Comment:
that's a nice method for fetching and checking the response

Reply:
thanks :)